From b2d19dc3f3b502469f6249c74e6f21c7ea77efa0 Mon Sep 17 00:00:00 2001
From: Olof Johansson <olof@ethup.se>
Date: Thu, 18 Jul 2013 14:14:46 +0200
Subject: [PATCH] checkbashisms: 'command' supports -vVp according to POSIX

POSIX mandates that the command command must support the three flags -p,
-v and -V. While -p was supported by checkbashisms, -v and -V was
explcititly labeled bashisms.

Signed-off-by: Olof Johansson <olof@ethup.se>
---
 scripts/checkbashisms.pl     |    1 -
 test/bashisms/command.sh     |    4 ++--
 test/bashisms/command.sh.out |    4 ----
 test/test_checkbashisms      |    2 +-
 4 files changed, 3 insertions(+), 8 deletions(-)
 delete mode 100644 test/bashisms/command.sh.out

diff --git a/scripts/checkbashisms.pl b/scripts/checkbashisms.pl
index 44369d4..a893e7d 100755
--- a/scripts/checkbashisms.pl
+++ b/scripts/checkbashisms.pl
@@ -636,7 +636,6 @@ sub init_hashes {
 	qr';;?&' =>  q<;;& and ;& special case operators>,
 	$LEADIN . qr'jobs\s' =>  q<jobs>,
 #	$LEADIN . qr'jobs\s+-[^lp]\s' =>  q<'jobs' with option other than -l or -p>,
-	$LEADIN . qr'command\s+-[^p]\s' =>  q<'command' with option other than -p>,
 	$LEADIN . qr'setvar\s' =>  q<setvar 'foo' 'bar' should be eval 'foo="'"$bar"'"'>,
 	$LEADIN . qr'trap\s+["\']?.*["\']?\s+.*(?:ERR|DEBUG|RETURN)' => q<trap with ERR|DEBUG|RETURN>,
 	$LEADIN . qr'(?:exit|return)\s+-\d' => q<exit|return with negative status code>,
diff --git a/test/bashisms/command.sh b/test/bashisms/command.sh
index ae60c97..2c9023f 100644
--- a/test/bashisms/command.sh
+++ b/test/bashisms/command.sh
@@ -2,5 +2,5 @@
 
 command test
 command -p test
-command -v test # BASHISM
-command -V test # BASHISM
+command -v test
+command -V test
diff --git a/test/bashisms/command.sh.out b/test/bashisms/command.sh.out
deleted file mode 100644
index 7a6b081..0000000
--- a/test/bashisms/command.sh.out
+++ /dev/null
@@ -1,4 +0,0 @@
-possible bashism in bashisms/command.sh line 5 ('command' with option other than -p):
-command -v test # BASHISM
-possible bashism in bashisms/command.sh line 6 ('command' with option other than -p):
-command -V test # BASHISM
diff --git a/test/test_checkbashisms b/test/test_checkbashisms
index a2d197d..37ef996 100755
--- a/test/test_checkbashisms
+++ b/test/test_checkbashisms
@@ -84,7 +84,7 @@ test_comments_in_quoted_strings2() {
 }
 
 test_command() {
-    found "bashisms/command.sh" "$(cat bashisms/command.sh.out)"
+    clean "bashisms/command.sh"
 }
 
 test_comments_parsing_fns() {
-- 
1.7.10.4

